#! /bin/sh
patch -p1 -l -f $* < $0
exit $?

From: Albert Astals Cid <aacid@kde.org>
Date: Mon, 24 Mar 2025 00:44:54 +0100
Subject: [PATCH 5/7] PSStack::roll: Protect against doing int = -INT_MIN

(cherry picked from commit d87bc726c7cc98f8c26b60ece5f20236e9de1bc3)

CVE-2025-32364
---
 poppler/Function.cc | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/poppler/Function.cc b/poppler/Function.cc
index d2be7d71..a902cdf3 100644
--- a/poppler/Function.cc
+++ b/poppler/Function.cc
@@ -1066,7 +1066,7 @@ void PSStack::roll(int n, int j)
     PSObject obj;
     int i, k;

-    if (unlikely(n == 0)) {
+    if (unlikely(n == 0 || j == INT_MIN)) {
         return;
     }
     if (j >= 0) {
--
2.47.2
